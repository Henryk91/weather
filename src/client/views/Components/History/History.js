import React, { Component } from 'react';
import { Detailed } from '../index';
// import './style.css';

export default class History extends Component {
  constructor(props) {
    super(props);
    this.state = {
      weatherData: null,
      searching: false
    };
    this.getDateInfo = this.getDateInfo.bind(this)
  }

  getDateInfo(e) {
    e.preventDefault()
    
    if (this.title.value) {
      const weatherData = this.props.weatherData;

      if(weatherData){
      this.setState({ search: true, weatherData: null })
      fetch(`/api/weather?coordinates=${weatherData.latitude},${weatherData.longitude},${this.title.value}T12:00:00`)
        .then(res => res.json())
        .then(data => {
 
          this.setState({ search: false, weatherData: data })
        });
      } else {
        alert("Please add location then press search or press the geo location button")
      }
    } else {
      alert("Please Add A Date")
    }

  }
  render() {

    return (
      <div >
        <h2>Pick a Date</h2>
        <form>
          <input id="datePick" ref={(c) => this.title = c} type="date" required="required"></input><br />
          <button id="dateSubmit" onClick={this.getDateInfo}>Submit</button>
          {this.state.search ? <div className="loader"></div> : null}
          {this.state.weatherData ? <Detailed weatherData={this.state.weatherData} /> : null}
        </form>
      </div>
    );
  }
}
