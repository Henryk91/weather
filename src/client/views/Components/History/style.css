#datePick{
  align-content: center;
  color: white;
  background: #007998;
}
#dateSubmit{
  font-size: 20px;
}
iframe{
  display: grid;
  height: 400px;
  width: 84vw;
  margin: 0px;
  margin-left: 8vw;
  padding: 0px;
}