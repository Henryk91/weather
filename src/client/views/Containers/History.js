import React, { Component } from 'react';
import DefaultDay from './DefaultDay'

export default class History extends Component {
  constructor(props) {
    super(props);
    this.state = {
      weatherData: null,
      searching: false
    };
    this.getDateInfo = this.getDateInfo.bind(this)
    // this.getData = this.getData.bind(this);
  }
  getDateInfo(e) {
    e.preventDefault()
    alert("hello" + this.title.value)
  }
  render() {

    return (
      <div >
        <h2>Pick a Date</h2>
        <form>
          <input id="datePick" ref={(c) => this.title = c} type="date"></input><br />
          <button onClick={this.getDateInfo}>Submit</button>
        </form>
      </div>
    );
  }
}
