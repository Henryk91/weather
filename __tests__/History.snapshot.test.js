
import History from '../src/client/views/Components/History/History';
import renderer from 'react-test-renderer';
import React from 'react';

import { weatherData } from "../testData/weatherData.sample"

test('History component renders with detaild and date checker', () => {

  const rendered = renderer.create(
    <History weatherData={weatherData()}/>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});