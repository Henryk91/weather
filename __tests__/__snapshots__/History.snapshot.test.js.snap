// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders with detaild and date checker 1`] = `
<div>
  <h2>
    Pick a Date
  </h2>
  <form>
    <input
      id="datePick"
      required="required"
      type="date"
    />
    <br />
    <button
      id="dateSubmit"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;
